TddShop/TddShop.Cli.Tests/Order/DiscountCalculatorTests.cs

```diff
@@ -4,12 +4,375 @@
+
+            // Assert
+            Assert.That(actual, Is.EqualTo(expected));
+        }
+
+        [Test]
+        public void GetDiscount_IfOrderPlacedInDecemberBuyOnlyLeftOvers_ShouldReturnLeftoversDiscounts()
+        {
+            // Arrange
+            var lastItemName = "Toy car";
+            var moreExpensiveLastItemName = "Expensive Cheese";
+
+            var orderModel = new OrderModel
+            {
+                CreatedAt = new DateTime(2016, 12, 13),
+                Items = new[]
+                {
+                    new ItemModel
+                    {
+                        Name = lastItemName,
+                        Price = 400,
+                    },
+                    new ItemModel
+                    {
+                        Name = moreExpensiveLastItemName,
+                        Price = 100,
+                    }
+                }
+            };
+
+            _stockRepositoryStub
+                .Setup(repository => repository.GetAvailableItemsByName(lastItemName))
+                .Returns(1);
+
+            _stockRepositoryStub
+                .Setup(repository => repository.GetAvailableItemsByName(moreExpensiveLastItemName))
+                .Returns(1);
+
+            var expected = 400 * 0.75M + 100 * 0.5M;
+
+            // Act
+            var actual = _target.GetDiscount(orderModel);
+
+            // Assert
+            Assert.That(actual, Is.EqualTo(expected));
+        }
+
+        [Test]
+        public void GetDiscount_IfOrderOnlyWithBeersFor5Eur_ShouldChooseBeerDiscount()
+        {
+            // Arrange
+            var beerCategory = "Beer";
+            var numberOfBeers = 100;
+            var beerPrice = 5;
+
+            var orderItems = Enumerable.Range(1, numberOfBeers).Select(i => new ItemModel
+            {
+                Category = beerCategory,
+                Price = beerPrice,
+            }).ToArray();
+
+            var orderModel = new OrderModel
+            {
+                Items = orderItems
+            };
+
+            var expected = numberOfBeers * (beerPrice - 2);
+
+            // Act
+            var actual = _target.GetDiscount(orderModel);
+
+            // Assert
+            Assert.That(actual, Is.EqualTo(expected));
+        }
+
+        [Test]
+        public void GetDiscount_IfOrderedInDecemberOnlyWithBeersFor5Eur_ShouldChooseDecemberDiscount()
+        {
+            // Arrange
+            var beerCategory = "Beer";
+            var numberOfBeers = 30;
+            var beerPrice = 5;
+
+            var orderItems = Enumerable.Range(1, numberOfBeers).Select(i => new ItemModel
+            {
+                Category = beerCategory,
+                Price = beerPrice,
+            }).ToArray();
+
+            var orderModel = new OrderModel
+            {
+                CreatedAt = new DateTime(2016, 12, 13),
+                Items = orderItems
+            };
+
+            var expected = numberOfBeers * (5 - 2);
```

Comment:
@michalczukm może lepiej dać już na stałe wyliczone te liczby w expected zamiast wyliczeń?

Reply:
zostawiłbym wyliczenia - widać z nich skąd się wzieły te wartości.
Ew zinlineował bym je do wywołania `Assert.That(...)`.

Co o tym sądzisz?

Reply:
tylko wtedy teoretycznie potrzebujemy testu do testu :P Zastanawiam się czy dać wartość sztywno z komentarzem skąd wyliczenie. ale nie upieram się, ot tak mi się w oczy rzuciło, zostawmy tak jak jest :)